src/xo-app/backup/new/index.js

```diff
@@ -183,31 +186,33 @@ export default class New extends Component {
     super(props)
 
     const { state } = this
-    const { job } = props
-
     state.cronPattern = '* * * * *'
+  }
 
-    if (!job) {
+  componentWillMount () {
+    const { job, schedule } = this.props
+    if ((!job && schedule) || (job && !schedule)) {
+      error(_('backupEditNotFoundTitle'), _('backupEditNotFoundMessage'))
+    }
+    if (!job || !schedule) {
       return
     }
+    this.setState({
+      backupInfo: BACKUP_METHOD_TO_INFO[job.method],
+      cronPattern: schedule.cron
+    }, () => delay(this._populateForm, 250, job)) // Work around.
+    // Without the delay, some selects are not always ready to load a value
+    // Values are displayed, but html5 compliant browsers say the value is required and empty on submit
+  }
 
+  _populateForm = (job) => {
     let values = getJobValues(job.paramsVector)
-    state.backupInfo = BACKUP_METHOD_TO_INFO[job.method]
-
     if (values.length !== 1) {
-      state.defaultValue = {
+      const { backupInput } = this.refs
+      backupInput.value = {
         ...getJobValues(values[1])[0],
         vms: getJobValues(values[0])
       }
-
-      return
-    }
-
-    // Older versions of XenOrchestra uses only values[0].
-    values = getJobValues(values[0])
-    state.defaultValue = {
-      ...values[0],
-      vms: map(values, value => value.id)
     }
```

Comment:
Sorry, I missed them.
It falls on me to be more sharp, but I'll notice 2 points :

- Old/New version are 2 cases, so for me a if/else would be better than an early return. I like to keep early returns for "interruption" conditions. (in fact I probably missed the lines because I mistook the `return }` surrounded by a lot of white space for the end of the function).

- As the "Old version" id the __particular__ case, it should be the case submitted to a condition, followed by an ending on the general case.

All this would have the same effect, but will provide a code more sticky to it's "sense".